### DISCUSSION TITLE
All PIMCore Ajax Requests get terminated after 30 seconds in PIMCore v10.5.19

### DISCUSSION THREAD

**Discussion by utkarsh-sharma-511** [pimcore/pimcore#15236] [Q&A / Support]
I am facing this issue in PIMCore v 10.5.19 in various environments. All PIMCore Ajax Requests on Objects like - tree-get-childs-by-id,grid-proxy, find requests fail if the request is not processed in 30 seconds for admin user. Does anyone know what can be done to overcome this? The 'max_execution_time' directive is set to 240 seconds. We also have redis pimcore.cache.pool. On nginx and .htaccess also we do not have any time limit or execution time restrictions. Attaching screenshots for reference. 

Any reference on what could be done to improve the performance or from where the requests are being restricted will help. 

![Screenshot from 2023-05-19 14-40-08](https://github.com/pimcore/pimcore/assets/88316749/df0771bc-bd0b-4e3b-b3f2-0f05427e7bf0)

![Screenshot from 2023-05-19 14-40-39](https://github.com/pimcore/pimcore/assets/88316749/a6694a7f-c688-4cd4-b844-f70b8ff8ae35)

![Screenshot from 2023-05-19 14-41-01](https://github.com/pimcore/pimcore/assets/88316749/bad922d3-6cac-483d-9057-e7baa228b498)

**Comment by pdchaudhary** [comment]
Check server configuration ensure that your server configuration, such as PHP settings, timeouts, and memory limits, are appropriately configured to handle requests.

**Comment by NiklasBr** [comment]
Default timeout on all actions in Pimcore is 30s due to [configuration in Ext JS](https://raw.githubusercontent.com/pimcore/pimcore/10.6/bundles/AdminBundle/Resources/public/extjs/js/ext-all-debug.js):

```js
        /**
         * @cfg {Number} [timeout=30000] The timeout in milliseconds to be used for
         * requests.
         * Defaults to 30000 milliseconds (30 seconds).
         *
         * When a request fails due to timeout the XMLHttpRequest response object will
         * contain:
         *
         *     timedout: true
         */
        timeout: 30000,
```
